### DISCUSSION TITLE
How can I find broken links within my documentation?

### DISCUSSION THREAD

**Discussion by nh916** [squidfunk/mkdocs-material#5760] [Q&A]
I am writing documentation with markdown files and python docstrings, how can I find all the broken links in the terminal and fix them?

**Comment by squidfunk** [comment]
There's the [`htmlproofer` plugin](https://github.com/manuzhang/mkdocs-htmlproofer-plugin) which I think does what you request. We plan to add support for external link validation to the privacy plugin in the future, as it already processes [external links](https://squidfunk.github.io/mkdocs-material/setup/ensuring-data-privacy/#external-links) and adding validation seems like a natural fit, but currently no ETA.

Internal link validation is a different story though.

**Comment by nh916** [comment]
I was having issues with both internal and external links as many of both were broken and I was not getting any notification for it in the terminal. I think this could have been for several reasons.
1. I was inputting links with html tags instead of md tags
2. I was inputting links into docstrings and generating documentation from there
3. I had another MD issue and it prevented from discovering broken links

I ended up using [Xenu's Link Checker](https://home.snafu.de/tilman/xenulink.html#Download) and [screaming frog](https://www.screamingfrog.co.uk/seo-spider/) to find all the broken links locally for now and fix them. 

However, this is more of a short term solution because ideally I want anything that breaks in documentation to immediately pop up in the CI for us to quickly catch and fix, so the user never sees a broken documentation

**Reply by squidfunk** [reply]
Jup, as said, we'll tackle external links (and maybe internal as well) later. In the meantime, using external services or third-party plugins is a good idea 😊
